passboltapi/__init__.py

```diff
@@ -225,32 +242,117 @@ def import_public_keys(self, trustlevel='TRUST_FULLY'):
     def describe_folder(self, folder_id):
         return self.get(f"/folders/{folder_id}.json")
 
-    def create_resource(self, name: str, username: str, password: str, uri: Optional[str] = None,
-                        description: Optional[str] = None, folder: Optional[str] = None):
-        """Creates a new resource on passbolt and shares it with the provided group and folder"""
-        r = self.post("/resources.json", {
+    def create_resource(self, name: str, password: str,
+                        username: str = "",
+                        description: str = "",
+                        uri: str = "",
+                        resource_type_id: PassboltResourceTypeIdType | None = None,
+                        folder: PassboltFolderTuple | None = None):
+        """Creates a new resource on passbolt and shares it with the provided folder recipients"""
```

Comment:
May be before doing the request we raise an exception if name and password are neither None nor empty string.